aries_cloudagent/connections/base_manager.py

```diff
@@ -57,6 +62,17 @@ def __init__(self, profile: Profile):
         self._logger = logging.getLogger(__name__)
         self._profile = profile
 
+        multitenant_mgr = profile.inject_or(BaseMultitenantManager)
+        wallet_id = profile.settings.get_str("wallet.id")
+        if multitenant_mgr and wallet_id:
+            self._route_manager: RouteManager = multitenant_mgr.get_route_manager(
+                profile, wallet_id
+            )
+        else:
+            self._route_manager: RouteManager = CoordinateMediationV1RouteManager(
+                profile
+            )
```

Comment:
I've never been a fan of how I implemented the multitenant routing part as it made subwallets aware of the hierarchy (i.e. it was the tenants responsiblity to call the base wallet to inform it about routing). 

With the approach you've taken here I think we could fully remove that dependency. What if we bind the route manager to the injection context and inject it here. That would mean the specific manager don't need to know about multitenancy at all as they're just injecting the route manager. Under the hood that could have a different implementation based on whether multitenancy is enabled or not.